From ebf79868a213561c62034970a4ddc9f3dee3b049 Mon Sep 17 00:00:00 2001
From: Michal Rostecki <mrostecki@suse.de>
Date: Mon, 29 Oct 2018 09:54:34 +0100
Subject: [PATCH] api: Change visibility of several api protos to public

Cilium extends Envoy by adding its own filter (which is needed for
proper Cilium-Envoy-Istio integration). In order to do that, Cilium
uses the following proto libraries[0][1]:

- @envoy_api//envoy/api/v2:discovery
- @envoy_api//envoy/api/v2/core:address
- @envoy_api//envoy/api/v2/core:config_source
- @envoy_api//envoy/api/v2/route:route

Those libraries had visibility set to "//visibility:private" or
":friends" which prevented Cilium from using them.

The following libraries are dependencies of libraries listed above,
so they have to be public too:

- @envoy_api//envoy/api/v2/core:base
- @envoy_api//envoy/api/v2/core:grpc_service

For now, Cilium is still using older version of Envoy which does
not specify visibility of api proto libraries, but upgrading Envoy
is impossible.

[0] https://github.com/cilium/cilium/blob/master/envoy/BUILD#L76-L78
[1] https://github.com/cilium/cilium/blob/master/envoy/BUILD#L109

Risk Level: low
Testing: n/a
Docs Changes: n/a
Release Notes: n/a

Signed-off-by: Michal Rostecki <mrostecki@suse.de>
---
 api/envoy/api/v2/BUILD       | 12 +++++++++---
 api/envoy/api/v2/core/BUILD  | 22 ++++++++++++++--------
 api/envoy/api/v2/route/BUILD | 11 ++++++++---
 3 files changed, 31 insertions(+), 14 deletions(-)

diff --git api/envoy/api/v2/BUILD api/envoy/api/v2/BUILD
index 261d14081..2f59677a1 100644
--- api/envoy/api/v2/BUILD
+++ api/envoy/api/v2/BUILD
@@ -1,4 +1,10 @@
-load("//bazel:api_build_system.bzl", "api_go_grpc_library", "api_go_proto_library", "api_proto_library_internal")
+load(
+    "//bazel:api_build_system.bzl",
+    "api_go_grpc_library",
+    "api_go_proto_library",
+    "api_proto_library",
+    "api_proto_library_internal",
+)
 
 licenses(["notice"])  # Apache 2
 
@@ -16,10 +22,10 @@ package_group(
     ],
 )
 
-api_proto_library_internal(
+api_proto_library(
     name = "discovery",
     srcs = ["discovery.proto"],
-    visibility = [":friends"],
+    visibility = ["//visibility:public"],
     deps = ["//envoy/api/v2/core:base"],
 )
 
diff --git api/envoy/api/v2/core/BUILD api/envoy/api/v2/core/BUILD
index 45251aebb..8d32005c1 100644
--- api/envoy/api/v2/core/BUILD
+++ api/envoy/api/v2/core/BUILD
@@ -1,4 +1,10 @@
-load("//bazel:api_build_system.bzl", "api_go_grpc_library", "api_go_proto_library", "api_proto_library_internal")
+load(
+    "//bazel:api_build_system.bzl",
+    "api_go_grpc_library",
+    "api_go_proto_library",
+    "api_proto_library",
+    "api_proto_library_internal",
+)
 
 licenses(["notice"])  # Apache 2
 
@@ -16,11 +22,11 @@ package_group(
     ],
 )
 
-api_proto_library_internal(
+api_proto_library(
     name = "address",
     srcs = ["address.proto"],
     visibility = [
-        ":friends",
+        "//visibility:public",
     ],
     deps = [":base"],
 )
@@ -31,11 +37,11 @@ api_go_proto_library(
     deps = [":base_go_proto"],
 )
 
-api_proto_library_internal(
+api_proto_library(
     name = "base",
     srcs = ["base.proto"],
     visibility = [
-        ":friends",
+        "//visibility:public",
     ],
     deps = [
         "//envoy/type:percent",
@@ -63,11 +69,11 @@ api_go_proto_library(
     deps = [":base_go_proto"],
 )
 
-api_proto_library_internal(
+api_proto_library(
     name = "config_source",
     srcs = ["config_source.proto"],
     visibility = [
-        ":friends",
+        "//visibility:public",
     ],
     deps = [
         ":base",
@@ -101,7 +107,7 @@ api_proto_library_internal(
     name = "grpc_service",
     srcs = ["grpc_service.proto"],
     visibility = [
-        ":friends",
+        "//visibility:public",
     ],
     deps = [":base"],
 )
diff --git api/envoy/api/v2/route/BUILD api/envoy/api/v2/route/BUILD
index 5bc601025..aa1c86f9f 100644
--- api/envoy/api/v2/route/BUILD
+++ api/envoy/api/v2/route/BUILD
@@ -1,11 +1,16 @@
-load("//bazel:api_build_system.bzl", "api_go_proto_library", "api_proto_library_internal")
+load(
+    "//bazel:api_build_system.bzl",
+    "api_go_proto_library",
+    "api_proto_library",
+    "api_proto_library_internal",
+)
 
 licenses(["notice"])  # Apache 2
 
-api_proto_library_internal(
+api_proto_library(
     name = "route",
     srcs = ["route.proto"],
-    visibility = ["//envoy/api/v2:friends"],
+    visibility = ["//visibility:public"],
     deps = [
         "//envoy/api/v2/core:base",
         "//envoy/type:range",
-- 
2.19.1

